### PR TITLE
Go-Control-Plane Version upgrade and remove deprecated fields

### LINKED ISSUE #1938: Fix Warning Logs within router due to envoy version upgrade to 1.18.2
### Description:
The following warning logs can be seen with router once the version is upgraded to 1.18.2. (After #1937)

It seems like it requires a new release from go-control-plane. 

```
router_1    | [2021-04-20 09:45:14.518][1][warning][misc] [source/common/protobuf/message_validator_impl.cc:21] Deprecated field: type envoy.config.core.v3.SubstitutionFormatString Using deprecated option 'envoy.config.core.v3.SubstitutionFormatString.text_format' from file substitution_format_string.proto. This configuration will be removed from Envoy soon. Please see https://www.envoyproxy.io/docs/envoy/latest/version_history/version_history for details. If continued use of this field is absolutely necessary, see https://www.envoyproxy.io/docs/envoy/latest/configuration/operations/runtime#using-runtime-overrides-for-deprecated-features for how to apply a temporary and highly discouraged override.
```

```
router_1    | [2021-04-20 09:45:11.765][1][warning][misc] [source/common/protobuf/message_validator_impl.cc:21] Deprecated field: type envoy.config.bootstrap.v3.Admin Using deprecated option 'envoy.config.bootstrap.v3.Admin.access_log_path' from file bootstrap.proto. This configuration will be removed from Envoy soon. Please see https://www.envoyproxy.io/docs/envoy/latest/version_history/version_history for details. If continued use of this field is absolutely necessary, see https://www.envoyproxy.io/docs/envoy/latest/configuration/operations/runtime#using-runtime-overrides-for-deprecated-features for how to apply a temporary and highly discouraged override.
```

### Environment details (with versions):
- OS:
- Client:
- Env (Docker/K8s):

---

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Purpose

To remove the deprecated fields from configurations, it was required to update the go-control-plane version

### Issues

Fixes #1938

### Automation tests
 - Unit tests added: No
 - Integration tests added: No (Covered by existing integration tests)

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/pkg/discovery/protocol/cache/v3/snapshot.go [hunk 1]
```diff
@@ -15,68 +15,59 @@
...
+		out.Resources[index] = NewResources(version, resource)
+	}
+
+	return out, nil
 }
 
-// GetResources selects snapshot resources by type.
-func (s *Snapshot) GetResources(typeURL string) map[string]types.Resource {
+// GetResources selects snapshot resources by type, returning the map of resources.
+func (s *Snapshot) GetResources(typeURL resource.Type) map[string]types.Resource {
```
Comment (VirajSalaka) [inline adapter/pkg/discovery/protocol/cache/v3/snapshot.go:54]: Check if this is no longer required.

File: router/src/main/resources/Dockerfile [hunk 2]
```diff
@@ -64,4 +64,4 @@ ENV ENVOY_GID=0
...
...ATH}'}, certificate_chain: {filename: '${ROUTER_PUBLIC_CERT_PATH}'}}, validation_context: {trusted_ca: {filename: '${ENFORCER_CA_CERT_PATH}'}}}}}, load_assignment: {cluster_name: access-logger, endpoints: [{lb_endpoints: [{endpoint: {address: {socket_address: {address: '${ENFORCER_ANALYTICS_HOST}', port_value: '${ENFORCER_ANALYTICS_RECEIVER_PORT}'}}}}]}]}}, {name: token_cluster, type: STRICT_DNS, connect_timeout: 20s, transport_socket: {name: envoy.transport_sockets.tls, typed_config: {'@type': type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext, common_tls_context: {tls_params: {tls_minimum_protocol_version: TLSv1_2, tls_maximum_protocol_version: TLSv1_2}, tls_certificates: {private_key: {filename: '${ROUTER_PRIVATE_KEY_PATH}'}, certificate_chain: {filename: '${ROUTER_PUBLIC_CERT_PATH}'}}, validation_context: {trusted_ca: {filename: '${ENFORCER_CA_CERT_PATH}'}}}}}, load_assignment: {cluster_name: token_cluster, endpoints: [{lb_endpoints: [{endpoint: {address: {socket_address: {address: '${ENFORCER_HOST}', port_value: 8082}}}}]}]}}]}, layeredRuntime: {layers: [{name: deprecation, staticLayer: {re2.max_program_size.error_level: 1000}}]} }" $TRAILING_ARGS
```
Comment (pubudu538) [inline router/src/main/resources/Dockerfile:67]: What is the change in here? I guess we are still writing the access logs to stdout..

**Comment by VirajSalaka** [comment]
https://github.com/wso2/product-microgateway/pull/2396#discussion_r749947665

The change was removing access_log_path for admin operations.
`admin: {access_log_path: /dev/null,`

This is for admin access logs. (logs are created when invoking admin port to do router manipulations). The only call we make is `http://localhost:9000/config_dump`.

**Review by pubudu538 — APPROVED** [review]

File: adapter/internal/discovery/xds/server.go [hunk 3]
```diff
@@ -736,9 +739,19 @@ func updateXdsCache(label string, endpoints []types.Resource, clusters []types.R
...
+		envoy_resource.ClusterType:  clusters,
+		envoy_resource.ListenerType: listeners,
+		envoy_resource.RouteType:    routes,
+	})
+	if err != nil {
+		logger.LoggerXds.Errorf("Error while updating the snapshot : %v", err.Error())
+		return false
+	}
+	err = snap.Consistent()
+	//TODO: (VirajSalaka) check
```
Comment (renuka-fernando) [inline adapter/internal/discovery/xds/server.go:753]: Is this TODO done? or have to check later

**Review by renuka-fernando — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[succeeded] dev-deployment : 20211202.25

**Pull request merged**